### DISCUSSION TITLE
Coupling phase field to mechanics

### DISCUSSION THREAD

**Discussion by sslim8537** [idaholab/moose#22791] [Q&A Getting Started] [answered]
Dear all,

I am new to Moose and just starting with the basic examples in the tutorials. After learning the basics, I will try to solve my problem, which is to simulate a multiphase field evolution coupled to the mechanics. From what I have read in the documentation, I think that the Moose framework is exactly what I need. However, before I invest a lot of time, I want to double check whether Moose is suitable for this task or not. I am wondering if Moose will allow for calculations:
    - on the three dimensional spherical domain,
    - of multiple-phase fields,
    - of coupled diffusion and phase boundary movement in the multi-phase field,
    - of coupling between phase field and mechanics (phase-field and stress field),
    - of concentration dependent diffusion coefficients
 
Thank you very much for your kind replies.

**Comment by GiudGiud** [comment, marked as answer]
Hello

MOOSE can do 1D, 2D, 3D XYZ, 2D RZ and 1D spherical. So a 3D spherical domain would work as long as it s represented in XYZ
Multiple phases is supported
Phase boundaries can move
Coupling between phase field and mechanics has been done. I believe you will find examples in MOOSE of this
Diffusion coefficients depending on concentration/local variables is fully supported

The phase field docs are here btw
https://mooseframework.inl.gov/modules/phase_field/index.html

@laagesen for checking what I say

Guillaume

**Reply by laagesen** [reply]
Yes agreed on all of this! The only thing I would add is that there are several models for multiple order parameters or multiple phase fields in the literature. We have the most support for the model described in 

https://journals.aps.org/pre/abstract/10.1103/PhysRevE.98.023309

And are currently revising documentation to give a more thorough description of how what's in the input files relates to the paper. This is a multi-order parameter model. There are other models that are generally referred to as multi-phase field models, the distinction being that those models enforce that the sum of all the phase-fields must equal 1 everywhere. There are pros and cons to the different approaches, which is a bit beyond the scope here, but the multi-phase model can also be implemented in MOOSE, we just don't have input files with examples in the repository.
